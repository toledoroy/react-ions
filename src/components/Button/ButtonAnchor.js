import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router'
import style from './style.scss'
import classNames from 'classnames/bind'

const ButtonAnchor = props => {
  const cx = classNames.bind(style)
  const collapseClass = props.collapse ? 'collapse' : null
  const btnAnchorClasses = cx(style.btn, props.optClass, props.size, collapseClass)

  let buttonAnchor

  const handleClick = e => {
    if (props.disabled) {
      e.preventDefault()
    }
  }

  if (props.internal) {
    buttonAnchor = <Link to={props.path} className={btnAnchorClasses} size={props.size} disabled={props.disabled} onClick={handleClick} style={props.style} className={props.className}>{props.children}</Link>
  } else {
    buttonAnchor = <a href={props.path} className={btnAnchorClasses} target={props.target} disabled={props.disabled} onClick={handleClick}>{props.children}</a>
  }
  return buttonAnchor
}

ButtonAnchor.propTypes = {
  /**
   * A string to allow for inline styles
   */
  style: PropTypes.string,
  /**
   * A class name to be used for local styles or integrations (required to support styled-components)
   **/
  className: PropTypes.string,
  /**
   * Optional styles to add to the button.
   */
  optClass: PropTypes.string,
  /**
   * The size of button.
   */
  size: PropTypes.string,
  /**
   * Whether the button is disabled.
   */
  disabled: PropTypes.bool,
  /**
   * A path to pass to the anchor tag.
   */
  path: PropTypes.string,
  /**
   * Whether the link it to an internal page, or external (default)
   */
  internal: PropTypes.bool,
  /**
   * Whether to display only an icon on small screens
   */
  collapse: PropTypes.bool
}

export default ButtonAnchor
