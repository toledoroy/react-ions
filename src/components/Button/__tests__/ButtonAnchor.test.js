import React from 'react'
import ButtonAnchor from '../ButtonAnchor'

describe('ButtonAnchor', () => {
  let wrapper

  it('displays a button anchor with an external link', () => {
    expect(shallow(<ButtonAnchor path='http://www.google.com' target='_blank'>External</ButtonAnchor>)).toMatchSnapshot()
  })

  it('displays a button anchor with an internal link', () => {
    expect(shallow(<ButtonAnchor path='/components/progress-bar' internal={true}>Internal</ButtonAnchor>)).toMatchSnapshot()
  })
})
