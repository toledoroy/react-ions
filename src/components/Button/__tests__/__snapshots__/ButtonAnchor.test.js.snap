// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAnchor displays a button anchor with an external link 1`] = `
<a
  className="btn"
  href="http://www.google.com"
  onClick={[Function]}
  target="_blank"
>
  External
</a>
`;

exports[`ButtonAnchor displays a button anchor with an internal link 1`] = `
<Link
  onClick={[Function]}
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/components/progress-bar"
>
  Internal
</Link>
`;
